### ISSUE TITLE
Violation on the closing parenthesis on a new line in an annotation array.

### ISSUE THREAD

**Issue by jdussouillez** [checkstyle/checkstyle#10582]
*Issue first reported in https://github.com/checkstyle/checkstyle/issues/10020#issuecomment-893156100*
*First analysis done by @Abhishek-kumar09 : https://github.com/checkstyle/checkstyle/issues/10020#issuecomment-895304847*

----

Since the v8.45, Checkstyle reports a false positive error when you close an annotation parenthesis on a new line if you're in an array. This happens only in an array parameter of an annotation, not in a method body.

*AnnotationArray.java*
```java
public class AnnotationArray {

    @APIResponses({
        // OK
        @APIResponse(responseCode = "200", description = "Success"),
        // OK
        @APIResponse(
            responseCode = "404",
            description = "Not found"),
        // KO
        @APIResponse(
            responseCode = "500",
            description = "Internal error"
        ) // Error is here (line 14)
    })
    public void get() {
        // OK
        var foo = List.of(
            1,
            2,
            3
        );
        // OK
        var bar = new Integer[]{
            1,
            2,
            3
        };
    }

    private static @interface APIResponse {
        String responseCode();
        String description();
    }

    private static @interface APIResponses {
        APIResponse[] value() default {};
    }
}
```

*checkstyle.xml*
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="error"/>

    <module name="SuppressWithPlainTextCommentFilter"/>
    <property name="fileExtensions" value="java, properties, xml"/>
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="severity" value="error"/>
            <property name="basicOffset" value="4"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="4"/>
            <property name="throwsIndent" value="4"/>
            <property name="lineWrappingIndentation" value="4"/>
            <property name="arrayInitIndent" value="4"/>
        </module>
    </module>
</module>
```

```shell
java -jar checkstyle-8.45-all.jar -c checkstyle.xml AnnotationArray.java
```

```
Starting audit...
[ERROR] /home/junior/Downloads/AnnotationArray.java:14:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

**Comment by nmancus1** [comment]
It appears that regression happened in 8.39 release:

```
➜  javac AnnotationArray.java 

➜  cat AnnotationArray.java 
import java.util.List;

public class AnnotationArray {

    @APIResponses({
        // OK
        @APIResponse(responseCode = "200", description = "Success"),
        // OK
        @APIResponse(
            responseCode = "404",
            description = "Not found"),
        // KO
        @APIResponse(
            responseCode = "500",
            description = "Internal error"
        ) // Error is here (line 16)
    })
    public void get() {
        // OK
        var foo = List.of(
            1,
            2,
            3
        );// OK
        var bar = new Integer[]{
            1,
            2,
            3
        };
    }

    private static @interface APIResponse {
        String responseCode();
        String description();
    }

    private static @interface APIResponses {
        APIResponse[] value() default {};
    }
}

➜  java -jar checkstyle-8.38-all.jar -c config.xml AnnotationArray.java
Starting audit...
Audit done.

➜  java -jar checkstyle-8.39-all.jar -c config.xml AnnotationArray.java
Starting audit...
[ERROR] AnnotationArray.java:16:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.

```

Regression could possibly be related to changes at https://github.com/checkstyle/checkstyle/issues/5951.

Issue is still valid:

```
➜  java -jar checkstyle-9.2-all.jar -c config.xml AnnotationArray.java
Starting audit...
[ERROR] AnnotationArray.java:16:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 1 errors.


```

**Comment by romani** [comment]
@Abhishek-kumar09 , it might be related to fix for https://github.com/checkstyle/checkstyle/issues/5951

**Comment by jdussouillez** [comment]
Still happening with v10.12.2

**Comment by kaiqkt** [comment]
Had this issue in v10.12.4

**Comment by wojciechcwek** [comment]
I have the same problem in 10.17.0. Are there plans to fix it?

**Comment by rnveach** [comment]
Anyone is welcome to work on a fix. There are no current plans to work on indentation issues.
